### PR TITLE
Blaze: manage campaigns: use `can_blaze` from me sites endpoint to check for blaze eligibility

### LINKED ISSUE #18704: Blaze: use `can_blaze` from me/sites endpoint to check for blaze eligibility
currently we rely on `/sites/{siteid}/blaze/status` to determine blaze eligibility.

update to using the new `can_blaze` property returned from `/me/sites` to determine eligibility.

slack ref: p1686836837070139/1686227996.772819-slack-c04ljfs1g5p
```[tasklist]
### Tasks
- [x] remove the api call for blaze status in wp android -https://github.com/wordpress-mobile/WordPress-Android/pull/18718
- [x] add the logic to check blaze eligibility from site model. - https://github.com/wordpress-mobile/WordPress-Android/pull/18718
- [x] update the logic of checking blaze eligibility on all entry points - dashboard| menu | page list | post list -https://github.com/wordpress-mobile/WordPress-Android/pull/18718
- [x] remove the data base and api classes  - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2762
- [x] remove the blaze status table from fluxC - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2762
```

### PR DESCRIPTION
**Pull request by AjeshRPai**
part of #18704

## Description 
This PR updates the eligibility check of a site to enable the blaze feature. 

## Explanation of changes 
- THis PR removes the blaze status API dependancy for the blaze eligibility check for a site - which was added with the fluxc PR  https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2657/commits
- this pr implements the blaze feature to be enabled by checking the `can_blaze` property in site model 

### test - blaze is not available

1. go to the **jetpack** app
2. 🚪 login with a wp account without administrator privileges or use a private site 
3. ⛳ go to **app settings** → **debug settings →** enable feature flag → `blaze`
4. 👉🏻go to dashboard
5. ✅ verify that the **🔥 promote with blaze** is not shown 

### test - blaze is available

1. go to the **jetpack** app
2. 🚪 login with a wp account with administrator privileges 
3. ⛳ go to **app settings** → **debug settings →** enable feature flag → `blaze`
4. 👉🏻go to dashboard
5. ✅ verify that the **🔥 promote with blaze** is shown

### Test - Blaze is not available

1. Go to the **Jetpack** app
2. 🚪 Login with a wp account without Administrator privileges or Use a private site 
3. 👉🏻Go to Pages → Published Pages 
4. 👆🏻Click on the more icon ⁞ → 
5. ✅ Verify that the **🔥 Promote with Blaze** is not shown 

### Test - Blaze is available

1. Go to the **Jetpack** app
2. 🚪 Login with a wp account with Administrator privileges 
3. 👉🏻Go to Pages → Published Pages 
4. 👆🏻Click on the more icon ⁞ → 
5. ✅ Verify that the **🔥 Promote with Blaze** is shown 

### Test - Blaze is not available
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account without Administrator privileges or Use a private site 
3. ⛳ Go to **app settings** → **Debug settings →** enable feature flag → `blaze`
4. 👉🏻Go to Posts → Published Posts 
5. 👆🏻Click on the more icon ⁞ → 
6. ✅ Verify that the **🔥 Promote with Blaze** is not shown 

### Test - Blaze is available
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account with Administrator privileges 
3. ⛳ Go to **app settings** → **Debug settings →** enable feature flag → `blaze`
4. 👉🏻Go to Posts → Published Posts 
5. 👆🏻Click on the more icon ⁞ → 
6. ✅ Verify that the **🔥 Promote with Blaze** is shown

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

## Merge instructions 
- Make sure that https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2762 is merged 
- update the fluxC version 
- remote donot merge label 
- merge as normal

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by zwarm — CHANGES_REQUESTED** [review]
@AjeshRPai - The blaze tests all pass as written 👍 ; however I am seeing issues with the refreshing indicator 😿 

Pixel 5, Android 13
Steps to recreate
- Fresh login
- Select site
- Navigate to dashboard
- The refresh indicator is never-ending
- Tap from Home > Menu > Home, the refresh indicator hides
- The PTR function no longer works
- Change sites and the PTR still is not functioning

**Comment by AjeshRPai** [comment]
Hey @zwarm 👋🏼 

Thanks for the review. The PTR issues has been fixed with the commit [acb4057](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/acb4057c44f133a611941dc3f5475ad715b448c0)

THe issue was that I was directly updating the mediator livedata without using the `postState` extension function which updates the `refresh` mediator live data value. 🤦🏼 . So have fixed the 
1. PTR refresh indicator never ending 
2. PTR not functioning

**Comment by AjeshRPai** [comment]
Hey @zwarm 

On Investigating the unit tests failure I found that 
1. the refresh in BlazeCardSource.kt is called twice so have done some changes with the commit [c245fda](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/c245fda5a484bd485253118c22f141070af43d41)
2. fixed the unit test according to the above logic 
3. get data logic is not inside coroutine, so changed that with [16c0fe2](https://github.com/wordpress-mobile/WordPress-Android/pull/18718/commits/16c0fe21d4f67e3d12e0a3b2f40f6715a53125e8)

I renamed the `promotewithBlazecardsource` to `blazecardsource`, on re-visiting this PR i felt that this renaming could have been shifted to another PR. I am sorry about that.  😞

File: WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt [hunk 1]
```diff
@@ -46,7 +46,9 @@ class BlazeFeatureUtils @Inject constructor(
     }
 
     fun isSiteBlazeEligible(siteModel: SiteModel): Boolean {
-        return siteModel.canBlaze && siteModel.isAdmin &&
+        return siteModel.canBlaze != null &&
+                siteModel.canBlaze &&
+                siteModel.isAdmin &&
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt:51]: @AjeshRPai - ❓ I thought we didn't need the isAdmin check, but it's quite likely that I am mistaken.

**Review by zwarm — APPROVED** [review]
@AjeshRPai - Thanks for tackling this. All the tests pass as expected. I left one question, but it is not a blocker. 🙇 

Have to wait on the FluxC PR hash before merging.

File: WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/blaze/BlazeFeatureUtils.kt:51]: Hey @zwarm 

I think We need the Admin check, checked IOS PR - [they have same logic](https://github.com/wordpress-mobile/WordPress-iOS/pull/20916/files). I think can_blaze returns true for contributor role as well, don't remember but I think I checked that as well. Feel free to test it again

**Pull request merged**